app/controllers/collection_numbers_controller.rb

```diff
@@ -14,16 +14,16 @@ class CollectionNumbersController < ApplicationController
   before_action :pass_query_params, except: :index
   before_action :store_location, except: [:index, :destroy]
 
-  def index # rubocop:disable Metrics/AbcSize
-    if params[:pattern].present? # rubocop:disable Style/GuardClause
-      collection_number_search and return
-    elsif params[:observation_id].present?
-      observation_index and return
-    elsif params[:by].present? || params[:q].present? || params[:id].present?
-      index_collection_number and return
-    else
-      list_collection_numbers and return
-    end
+  @dispatch_table_for_index_subactions = {
+    pattern: :collection_number_search,
+    observation_id: :observation_index,
+    by: :index_collection_number,
+    q: :index_collection_number,
+    id: :index_collection_number
+  }.freeze
+
+  def index
```

Comment:
Question: Should I try to completely push up `index` to the superclass (leaving this file without an explicit `index` method). 
Advantage: It would be DRYer.
I left it this way because you don't have to look at another file to understand how `index` works here.